e3sm_diags/driver/arm_diags_driver.py

```diff
@@ -26,168 +30,165 @@
+                ds_test = test_ds.get_time_series_dataset(var, single_point=True)
+                test_diurnal, lst = composite_diurnal_cycle(  # type: ignore
+                    ds_test, var, season, fft=False
                 )
 
-                parameter.viewer_descr[var] = test[var].long_name
-                # Get the name of the data, appended with the years averaged.
-                parameter.test_name_yrs = test_data.get_name_yrs_attr()
-                parameter.var_name = test[var].long_name
-                parameter.var_units = test[var].units
-
                 refs = []
 
                 if "armdiags" in ref_name:
                     if region != "sgpc1":
-                        msg = "Diurnal cycle of {} at Site: {} is not supported yet".format(
-                            region, var
+                        raise RuntimeError(
+                            f"Diurnal cycle of {region} at Site: {var} is not "
+                            "supported yet"
                         )
-                        raise RuntimeError(msg)
                     else:
                         ref_file_name = "sgparmdiagsmondiurnalC1.c1.nc"
-
                         ref_file = os.path.join(ref_path, ref_file_name)
-                        ref = xr.open_dataset(ref_file)
+                        ds_ref = xr.open_dataset(ref_file)
+
                         if var == "PRECT":
-                            ref[var] = (
-                                ref["pr"] * 3600.0 * 24
-                            )  # Converting mm/second to mm/day"
-                            ref["lat"] = test.lat.values
-                            ref["lon"] = test.lon.values
-                            ref_diurnal, lst = composite_diurnal_cycle(
-                                ref, var, season, fft=False
+                            # Converting mm/second to mm/day"
+                            ds_ref[var] = ds_ref["pr"] * 3600.0 * 24
+                            ds_ref["lat"] = ds_test.lat.values
+                            ds_ref["lon"] = ds_test.lon.values
+
+                            ref_diurnal, lst = composite_diurnal_cycle(  # type: ignore
+                                ds_ref, var, season, fft=False
                             )
 
                             ref = ref_diurnal
 
                 else:
                     ref_data = Dataset(parameter, data_type="ref")
-                    # ref is a dataset
-                    ref = ref_data.get_time_series_dataset(var, single_point=True)
-                    ref_diurnal, lst = composite_diurnal_cycle(
-                        ref, var, season, fft=False
+                    ds_ref = ref_data.get_time_series_dataset(var, single_point=True)
+
+                    # FIXME: ref is not set to ref_diurnal here
+                    ref_diurnal, lst = composite_diurnal_cycle(  # type: ignore
+                        ds_ref, var, season, fft=False
                     )
```

Comment:
I noticed that `ref` is never updated to `ref_diurnal` here. Should this be fixed?